test/controllers/claimToAdjustPoa/SelectYourReasonControllerSpec.scala

```diff
@@ -0,0 +1,232 @@
+        status(result) shouldBe INTERNAL_SERVER_ERROR
+      }
+
+      "Something goes wrong in payment on account session Service" in {
+        setupTest(
+          sessionResponse = Left(new Exception("Something went wrong")),
+          claimToAdjustResponse = poa)
+
+        val result = TestSelectYourReasonController.show(isAgent = false, isChange = false)(fakeRequestWithNinoAndOrigin("PTA"))
+
+        status(result) shouldBe INTERNAL_SERVER_ERROR
+      }
+    }
+  }
+
+  "SelectYourReasonController.submit" should {
+
+    s"return $BAD_REQUEST" when {
+
+      s"no option is selected" in {
+
+        setupTest(
+          sessionResponse = Right(Some(PoAAmendmentData())),
+          claimToAdjustResponse = poa)
+
+        val request = FakeRequest(POST, routes.SelectYourReasonController.submit(isAgent = false, isChange = false).url)
+          .withSession("nino" -> BaseTestConstants.testNino, "origin" -> "PTA")
+
+        val result = TestSelectYourReasonController.submit(isAgent = false, isChange = false)(request)
+
+        status(result) shouldBe BAD_REQUEST
+      }
+    }
+
+    s"return $SEE_OTHER" when {
+
+      "in normal mode" when {
+
+        "if 'totalAmount' is equal to or greater than 'poaRelevantAmount'" in {
+
+          setupTest(
+            sessionResponse = Right(Some(PoAAmendmentData())),
+            claimToAdjustResponse = poa)
+
+          setupMockPaymentOnAccountSessionServiceSetAdjustmentReason(MainIncomeLower)
+
+          val request = FakeRequest(POST, routes.SelectYourReasonController.submit(isAgent = false, isChange = false).url)
+            .withFormUrlEncodedBody("value" -> "MainIncomeLower")
+            .withSession("nino" -> BaseTestConstants.testNino, "origin" -> "PTA")
+
+          val result = TestSelectYourReasonController.submit(isAgent = false, isChange = false)(request)
+
+          status(result) shouldBe SEE_OTHER
+          redirectLocation(result) shouldBe Some("/report-quarterly/income-and-expenses/view/adjust-poa/enter-poa-amount")
+        }
+
+        "if 'totalAmount' is less than 'poaRelevantAmount'" in {
+          setupTest(
+            sessionResponse = Right(Some(PoAAmendmentData(newPoAAmount = Some(5000.0)))),
+            claimToAdjustResponse = poaTotalLessThanRelevant)
+
+          setupMockPaymentOnAccountSessionServiceSetAdjustmentReason(MainIncomeLower)
+
+          val request = FakeRequest(POST, routes.SelectYourReasonController.submit(isAgent = false, isChange = false).url)
+            .withFormUrlEncodedBody("value" -> "MainIncomeLower")
+            .withSession("nino" -> BaseTestConstants.testNino, "origin" -> "PTA")
+
+          val result = TestSelectYourReasonController.submit(isAgent = false, isChange = false)(request)
+
+          status(result) shouldBe SEE_OTHER
+          redirectLocation(result) shouldBe Some("/report-quarterly/income-and-expenses/view/adjust-poa/check-your-answers")
+        }
+      }
+
+      "in check mode" when {
```

Comment:
I would assume this UTs is still something in TODO?

Reply:
I think there's a ticket for doing CYA / change answers.

Reply:
If we can refer to specific story in these TODO, as its a bit confusing at the moment and we can easily forgot about these in the future; there are quite a few TODO on line 104 and 108 in this file